### PR TITLE
[DUOS-1674]Update Algorithms.md with V3

### PR DESCRIPTION
**Pull request by kaitlynoflaherty**
## Addresses
https://broadworkbench.atlassian.net/jira/software/c/projects/DUOS/boards/123?modal=detail&selectedIssue=DUOS-1674

Updating Algorithm V3 documentation

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: docs/Algorithms.md [hunk 1]
```diff
@@ -153,3 +153,188 @@ as developed for the GA4GH as well as Disease Codes (**DS-X**) from the [Human D
 		</tr>	
 	</tbody>
 </table>
+
+## Version 3
```
Comment (rushtong) [inline docs/Algorithms.md]: Let's add some UI enhancements here. We should have v1, v2 collapsed and v3 expanded. See https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections for how to do that.

Also, let's move v3 to the top, v2 next, then v1 last.

File: docs/Algorithms.md [hunk 2]
```diff
@@ -3,24 +3,202 @@
...
 of the Matching APIs. This document describes the different approaches to matching a Research Purpose
 and a Consent on a dataset. Both Purpose and Consent are structured objects which allows for computational
-matching at scale. 
+matching at scale.
 
-## Version 1
-### Deprecated & Removed
-The original version of the algorithm uses an ontology tree to match a purpose and consent. First, we 
-construct a composite ontology tree from:
+## Version 3
```
Comment (fboulnois) [inline docs/Algorithms.md:8]:
```suggestion
## Version 3

```

File: docs/Algorithms.md [hunk 3]
```diff
@@ -3,24 +3,202 @@
...
+<details>
+<summary>View</summary>
```
Comment (fboulnois) [inline docs/Algorithms.md:11]:
```suggestion
<summary>View</summary>

```

File: docs/Algorithms.md [hunk 4]
```diff
@@ -3,24 +3,202 @@
...
+</table>
 
-Next, we create ontology nodes for the consent, the purpose and add them to the composite tree. Using 
-an [OWL](https://github.com/owlcs/owlapi) Reasoner, we determine if the purpose is a subclass of the 
-consent, or not. A valid ontological subclass for a research purpose indicates a successful match 
-between the purpose and the consent. See [Use Restriction Grammar](./UseRestrictionGrammar.md) for how
-we create ontology nodes for a consent or research purpose.
+### Abstain from Decision
+
+Describes the involvement of studies that <strong>will not</strong> result in a decision from the DUOS algorithm.
```
Comment (fboulnois) [inline docs/Algorithms.md]:
```suggestion
Describes the involvement of studies that <strong>will not</strong> result in a decision from the DUOS algorithm.

```

File: docs/Algorithms.md [hunk 5]
```diff
@@ -3,24 +3,202 @@
...
+      <td>
+        <li>NA</li>
+      </td>
+    </tr>
+  <tbody>
+</table>
+
+</details>
 
 ## Version 2
```
Comment (fboulnois) [inline docs/Algorithms.md:128]:
```suggestion
## Version 2

```

File: docs/Algorithms.md [hunk 6]
```diff
@@ -3,24 +3,202 @@
...
+<details>
+<summary>View</summary>
```
Comment (fboulnois) [inline docs/Algorithms.md:131]:
```suggestion
<summary>View</summary>

```

File: docs/Algorithms.md [hunk 7]
```diff
@@ -153,3 +331,23 @@ as developed for the GA4GH as well as Disease Codes (**DS-X**) from the [Human D
 		</tr>	
 	</tbody>
 </table>
+</details>
+
+## Version 1
```
Comment (fboulnois) [inline docs/Algorithms.md:267]:
```suggestion
## Version 1

```

File: docs/Algorithms.md [hunk 8]
```diff
@@ -153,3 +331,23 @@ as developed for the GA4GH as well as Disease Codes (**DS-X**) from the [Human D
...
+<details>
+<summary>View</summary>
+
+### Deprecated & Removed
```
Comment (fboulnois) [inline docs/Algorithms.md:272]:
```suggestion
### Deprecated & Removed

```

**Review by fboulnois — APPROVED** [review]
looks good, minor spacing nits:

File: docs/Algorithms.md [hunk 9]
```diff
@@ -3,24 +3,207 @@
...
+        <details>
+        <summary> Any of the following</summary>
+					<li>Limitation to one gender</li>
+          <li>Restricted to a pediatric population (under the age of 18)</li>
+					<li>Illegal behaviors (violence, domestic abuse, prostitution, sexual victimization)</li>
+          <li>Alcohol or drug abuse, or abuse of other addictive products</li>
+          <li>Sexual preferences or sexually transmitted diseases</li>
+          <li>Any stigmatizing illnesses</li>
+          <li>Vulnerable populations as defined in 456 CFR (children, prisoners, pregnant women, mentally disabled persons, or ["SIGNIFICANTLY"] economically or educationally disadvantaged persons)</li>
+          <li>Population Origins/Migration patterns (Answer is pre-determined based on selections made in Step 2)</li>
```
Comment (rushtong) [inline docs/Algorithms.md]: In this context, there is no form to reference
```suggestion
          <li>Population Origins/Migration patterns</li>
```

File: docs/Algorithms.md [hunk 10]
```diff
@@ -3,24 +3,207 @@
...
+</table>
+
+### Abstain from Decision
+
+Describes the involvement of studies that <strong>will not</strong> result in a decision from the DUOS algorithm.
+
+<table>
+	<thead>
+		<tr>
+			<th>If the DAR aims to involve...</th>
```
Comment (rushtong) [inline docs/Algorithms.md]: Here and throughout - in this context, we are describing a Research Purpose, not specifically a DAR. 
```suggestion
			<th>If the Research Purpose aims to involve...</th>
```

File: docs/Algorithms.md [hunk 11]
```diff
@@ -3,24 +3,207 @@
...
+          <li>Geographical Restrictions (i.e. <strong>GS-</strong>)</li>
+          <li>Public Moratorium/Embargo (i.e. <strong>MOR</strong>)</li>
+        </ul>	
+			</td>
+      <td>
+        <li>GS-, MOR are poorly defined and infrequently (never)</li>
+        <li>More detail/data needed</li>
+      </td>
+      <td>
+        <li>No requirement in DAR or Algorithm</li>
```
Comment (rushtong) [inline docs/Algorithms.md]: The language here doesn't make sense to me. I think we can remove this altogether if there are no DUO terms to describe it.

File: docs/Algorithms.md [hunk 12]
```diff
@@ -3,24 +3,207 @@
...
+				<ul>
+          <li>Genetic Studies Only (i.e. <strong>GSO</strong>)</li>
+          <li>Publication Required (i.e. <strong>PUB</strong>)</li>
+        </ul>	
+			</td>
+      <td>
+        <li>Not used in algorithmic decision-making</li>
+      </td>
+      <td>
+        <li>Assent in the DAR form</li>
```
Comment (rushtong) [inline docs/Algorithms.md]: Similar concern here - we should remove this bullet point

File: docs/Algorithms.md [hunk 13]
```diff
@@ -3,24 +3,207 @@
...
+    <tr>
+			<td>
+				<ul>
+          <li>Collabration Required (i.e. <strong>COL</strong>)</li>
+          <li>Ethics Approval Required (i.e. <strong>IRB</strong>)</li>
+        </ul>	
+			</td>
+      <td>
+        <li>DAR includes uploaded documents</li>
+        <li>Upload in the DAR form results in trigger a bypass of the algorithm</li>
```
Comment (rushtong) [inline docs/Algorithms.md]: Both cases here refer to something outside the scope of an algorithm discussion. I think we should remove them.

File: docs/Algorithms.md [hunk 14]
```diff
@@ -3,24 +3,207 @@
...
+          <li><strong>Deny</strong> if the dataset's Primary DUO terms are DS-, POA</li>
+				</ul>
+			</td>
+		</tr>
+	</tbody>
+</table>
+
+### Abstain from Decision
+
+Describes the involvement of studies that <strong>will not</strong> result in a decision from the DUOS algorithm.
```
Comment (rushtong) [inline docs/Algorithms.md]: After considering the abstention columns, I think we're adding more complexity and confusion with the second and third columns. I would suggest removing them altogether and adding some content here that explains a little more about why we are abstaining. Something along the lines of ...

> Due to the variety of sensitive research areas, ethical reasons, and areas where categorization is not possible, the DUOS system <strong>will not</strong> render a decision in the following cases.

**Review by rushtong — CHANGES_REQUESTED** [review]

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

**Pull request merged**